### DISCUSSION TITLE
Unable to compile on Arch Linux

### DISCUSSION THREAD

**Discussion by System64fumo** [KRTirtho/spotube#1041] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When trying to build by following the [tutorial](https://github.com/KRTirtho/spotube/blob/master/CONTRIBUTION.md#your-first-code-contribution) i'm unable to run the ``dart run build_runner...``
I get

```
Resolving dependencies in /tmp/spotube... 
Because spotube requires the Flutter SDK, version solving failed.

Flutter users should run `flutter pub get` instead of `dart pub get`.
```

### Expected Behavior

Building succesfully

### Steps to reproduce

1: git clone https://github.com/KRTirtho/spotube
2: cd spotube
3: cp .env.example .env (Then editing it)
4: flutter pub get && dart run build_runner build --delete-conflicting-outputs --enable-experiment=records,patterns

### Operating System

Arch Linux

### Spotube version

3.4.0

### Installation source

Website (spotube.netlify.app) or (spotube.krtirtho.dev)

### Additional information

I'm Trying to build it from source since there is no aarch64 binary available yet (at least in the releases section or AUR), And i've seen discussions about it before in previous threads [this one](https://github.com/KRTirtho/spotube/issues/715)
Thing is.. I can't even build it on my x86_64 system..
Surely i'm doing something wrong here since i never really compiled a flutter/dart program before.
Any help would be appreciated.

**Comment by KRTirtho** [comment]
Which version of Flutter are you using? Make sure it's the latest (16.x)

**Reply by KRTirtho** [reply]
Also `/tmp` not an idea directory for this kind of operation. Instead use ~/Downloads or any other user folder

**Reply by System64fumo** [reply]
I'm using flutter version 3.16.5 with dart 3.2.3

As for /tmp, I always build in tmpfs and never had any issue with it.
Unrelated to this, I've also tried building it in my home directory and got the same result

**Comment by ghost** [comment]
for some reason running dart from different path solved my problem:
```
$ whereis dart
dart: /usr/bin/dart /opt/flutter/bin/dart /opt/dart-sdk/bin/dart

$ /opt/flutter/bin/dart run build_runner build --delete-conflicting-outputs --enable-experiment=records,patterns
[...]
[INFO] Succeeded after 26.3s with 262 outputs (1555 actions)
```
OS: Arch Linux x86_64
Kernel: 6.7.9-arch1-1
Shell: bash 5.2.26
